src/tests/ftest/nvme/enospace.py

```diff
@@ -0,0 +1,448 @@
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: To validate DER_NOSPACE for SCM and NVMe
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a NvmeEnospace object."""
+        super(NvmeEnospace, self).__init__(*args, **kwargs)
+        self.daos_cmd = None
+
+    def setUp(self):
+        super(NvmeEnospace, self).setUp()
+
+        # initialize daos command
+        self.daos_cmd = DaosCommand(self.bin)
+        self.create_pool_max_size()
+        self.der_nospace_count = 0
+        self.other_errors_count = 0
+
+    def der_enspace_log_count(self):
+        """
+        Function to count the DER_NOSPACE and other ERR in client log.
+        """
+        #Get the Client side Error from client_log file.
+        output = []
+        cmd = 'cat {} | grep ERR'.format(get_log_file(self.client_log))
+        task = run_task(self.hostlist_clients, cmd)
+        for buf, _nodes in task.iter_buffers():
+            output = str(buf).split('\n')
+
+        for line in output:
+            if 'ERR' in line:
+                if 'DER_NOSPACE' in line:
+                    self.der_nospace_count += 1
+                else:
+                    self.other_errors_count += 1
+
+    def verify_enspace_log(self, der_nospace_err_count):
+        """
+        Function to verify there are no other error except DER_NOSPACE
+        in client log and also DER_NOSPACE count is higher.
+
+        args:
+            expected_err_count(int): Expected DER_NOSPACE count from client log.
+        """
+        #Get the DER_NOSPACE and other error count from log
+        self.der_enspace_log_count()
+
+        #Check there are no other errors in log file
+        if self.other_errors_count > 0:
+            self.fail('Found other errors, count {} in client log {}'
+                      .format(self.other_errors_count, self.client_log))
+        #Check the DER_NOSPACE error count is higher if not test will FAIL
+        if self.der_nospace_count < der_nospace_err_count:
+            self.fail('Expected DER_NOSPACE should be > {} and Found {}'
+                      .format(der_nospace_err_count, self.der_nospace_count))
+
+    def delete_all_containers(self):
+        """
+        Delete all the containers.
+        """
+        #List all the container
+        kwargs = {"pool": self.pool.uuid, "svc": self.pool.svc_ranks}
+        continers = (self.daos_cmd.get_output("pool_list_cont", **kwargs))
+     
+        #Destroy all the containers
+        for _cont in continers:
+            kwargs["cont"] = _cont
+            self.daos_cmd.container_destroy(**kwargs)
+
+    def ior_bg_thread(self, results):
+        """Start IOR Background thread, This will write small data set and
+        keep reading it in loop until it fails or main program exit.
+
+        Args:
+            results (queue): queue for returning thread results
+        """
+        con_uuid = str(uuid.uuid4())

```

Comment:
I couldn't tell clearly if you are letting ior create container or you are creating them using daos utility. But if you are letting ior do so, I would recommend, creating your own containers. That is more recommended way by Mohamad as per my conversations with him. He might remove that bit in future so better to have our own containers created.

Reply:
Updated the code to create the container first and use the container uuid for background job

Reply:
> IOR is creating the container I am only creating the UUID and allow it to use by IOR so later same UUID can be used for reading purpose. You mean IOR will not create the container ?

IOR creates container for now, but it's preferred we create our own container and supply it to the ior run command as it can be deprecated as well in future.